scripts/gas-refund-program/computeMerkleTree.ts

```diff
@@ -67,37 +68,193 @@ export async function computeAndStoreMerkleTreeForChain({
           }
         : {}),
     },
-    attributes: [
-      'address',
-      [
-        Sequelize.fn('SUM', Sequelize.col('refundedAmountPSP')),
-        'refundedAmountPSP',
-      ],
-    ],
-    group: ['address'],
+    attributes: ['address', 'timestamp', 'refundedAmountPSP', 'chainId'],
     raw: true,
   });
+}
 
-  const merkleTree = await computeMerkleData({
-    chainId,
+export async function computeAndStoreMerkleTree(epoch: number) {
+  const userRewardsOnStakingChains = await computeStakingChainsRefundedAmounts(
     epoch,
-    refundableTransactions,
-  });
+  );
+
+  const userRewards: AddressRewards[] = Object.keys(userRewardsOnStakingChains)
+    .map(account =>
+      Object.entries(userRewardsOnStakingChains[account]).map(
+        ([chainId, { amount, breakDownGRP }]) => ({
+          chainId: +chainId,
+          amount,
+          breakDownGRP,
+          account,
+        }),
+      ),
+    )
+    .flat()
+    .filter(entry => !entry.amount.eq(0));
+
+  const userGRPChainsBreakDowns = userRewards.reduce<AddressRewardsMapping>(
+    (acc, curr) => {
+      acc[curr.account] = curr.breakDownGRP;
```

Comment:
this `breakDownGRP` in json files, it represents what? total refunded by chain by the end of the epoch based on the stake chain the JSON is generated for?

If so, there's a flaw: both "1.json" and "10.json" end up with optimism's data in them, because optimism breakdown overrides ethereum's one in this code:

debugger breakpoint for the context:
<img width="1669" alt="Screenshot 2023-09-07 at 10 36 46" src="https://github.com/paraswap/paraswap-volume-tracker/assets/60445720/b0928113-20e6-4040-9227-38d268048137">


Json files:

<img width="863" alt="Screenshot 2023-09-07 at 10 39 42" src="https://github.com/paraswap/paraswap-volume-tracker/assets/60445720/27b6659b-6de3-4dc8-a9d2-6f572657649f">
<img width="676" alt="Screenshot 2023-09-07 at 10 39 11" src="https://github.com/paraswap/paraswap-volume-tracker/assets/60445720/87d79be4-0a7d-4f92-b59b-d2c44318e968">

Reply:
I fixed it [here](https://github.com/paraswap/paraswap-volume-tracker/commit/ffd2723b987dcdd29d14b68bfabe41f7ecfd8d8e)
hope that makes sense